From 198b4d93050103f6ec0d585a54f728105752d7e6 Mon Sep 17 00:00:00 2001
From: Joakim Zhang <qiangqing.zhang@nxp.com>
Date: Thu, 21 Apr 2022 22:17:53 +0800
Subject: [PATCH 051/274] MLK-25919-2 net: can: flexcan: don't jump out probe
 when setup stop-mode failed

User don't declare stop mode property in the device tree is not fatal,
even the soc support it, so should not break device probe for such case.

Signed-off-by: Joakim Zhang <qiangqing.zhang@nxp.com>
Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Reviewed-by: Jun Li <jun.li@nxp.com>
---
 drivers/net/can/flexcan.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/net/can/flexcan.c b/drivers/net/can/flexcan.c
index 18d7bb99e..9951b456d 100644
--- a/drivers/net/can/flexcan.c
+++ b/drivers/net/can/flexcan.c
@@ -2283,8 +2283,7 @@ static int flexcan_probe(struct platform_device *pdev)
 	err = flexcan_setup_stop_mode(pdev);
 	if (err < 0) {
 		if (err != -EPROBE_DEFER)
-			dev_err(&pdev->dev, "setup stop mode failed\n");
-		goto failed_setup_stop_mode;
+			dev_warn(&pdev->dev, "setup stop mode failed\n");
 	}
 
 	of_can_transceiver(dev);
@@ -2292,8 +2291,6 @@ static int flexcan_probe(struct platform_device *pdev)
 
 	return 0;
 
- failed_setup_stop_mode:
-	unregister_flexcandev(dev);
  failed_register:
 	pm_runtime_put_noidle(&pdev->dev);
 	pm_runtime_disable(&pdev->dev);
-- 
2.25.1

